src/aws-api-mcp-server/awslabs/aws_api_mcp_server/core/common/cloudwatch_logger.py

```diff
@@ -0,0 +1,152 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import boto3
+import threading
+import time
+from ..aws.credentials import get_local_credentials
+from ..common.config import AWS_API_MCP_PROFILE_NAME
+from botocore.exceptions import ClientError
+from loguru import logger
+from typing import Optional
+
+
+class CloudWatchLogSink:
+    """File-like sink that forwards log lines to CloudWatch Logs.
+
+    Notes:
+    - Uses a dedicated log stream per process/host.
+    - Maintains and refreshes the sequence token on InvalidSequenceToken errors.
+    - Thread-safe via an internal lock.
+    """
+
+    def __init__(self, log_group_name: str, region_name: str):
+        """Initialize the CloudWatchLogSink."""
+        self.log_group_name = log_group_name
+        self.region_name = region_name
+        self.lock = threading.Lock()
+        self.sequence_token: Optional[str] = None
+
+        credentials = get_local_credentials(profile=AWS_API_MCP_PROFILE_NAME)
+
+        self.client = boto3.client(
+            'logs',
+            region_name=region_name,
+            aws_access_key_id=credentials.access_key_id,
+            aws_secret_access_key=credentials.secret_access_key,
+            aws_session_token=credentials.session_token,
+        )
+
+        self.log_stream_name = f'aws-api-mcp-server-{str(int(time.time()))}'
+
+        self._ensure_log_group_exists()
+        self._ensure_log_stream_exists()
+        self._refresh_sequence_token()
+
+    def write(self, message: str):
+        """Write a log message to CloudWatch Logs."""
+        if not message:
+            return
+
+        # CloudWatch Logs requires event timestamps in milliseconds
+        timestamp_ms = int(time.time() * 1000)
+        log_event = {'timestamp': timestamp_ms, 'message': message.rstrip('\n')}
+
+        with self.lock:
+            self._put_log_events([log_event])
```

Comment:
The lock is not the problem as much as the call to `PutLogEvents` being potentially blocking - have we run a "load test" of sorts to figure out what the average latency for `PutLogEvents` is?